lang/cs/Org.Apache.REEF.Tests/Functional/FaultTolerant/PoisonTest.cs

```diff
@@ -108,6 +112,20 @@ public void OnNext(IActiveContext value)
             public void OnNext(IFailedEvaluator value)
             {
                 Logger.Log(Level.Error, FailedEvaluatorMessage);
+                if (value.FailedTask.Value == null)
+                {
+                    // TODO[JIRA REEF-1343]: fail the test if there's no failed task
+                    Logger.Log(Level.Error, "No failed task associated with failed evaluator");
+                }
+                else
+                {
+                    Logger.Log(Level.Error, "Failed task id '" + value.FailedTask.Value.Id + "'");
+                }
+            }
+            public void OnNext(ICompletedTask value)
+            {
+                // TODO[JIRA REEF-1343]: fail the test if receive ICompletedTask after failed evaluator
+                Logger.Log(Level.Info, "ICompletedTask");
```

Comment:
No, validation done in AppVeyor is exactly the same as done locally. And yes, driver should be able to ignore any events that happen. That's why I'm saying there's a bug here, probably more than one (one, task shouldn't complete, and two, presence of handler shouldn't be critical if its action is not verified).